### ISSUE TITLE
MacOS: dylib not found

### ISSUE THREAD

**Issue by jayjah** [tmthecoder/dargon2#1]
It's a standalone dart project, so no flutter sdk involved.

Host: MacOSX 10.15.6 (19G2021)
dart: Dart SDK version: 2.10.4 

Error:

```
  Invalid argument(s): Failed to load dynamic library (dlopen(lib/src/blobs/libargon2-darwin.dylib, 1): no suitable image found.  Did find:
        file system relative paths not allowed in hardened programs)
  dart:ffi                                                       new DynamicLibrary.open
  package:dargon2/src/native/loaders/dart_lib_loader.dart 20:27  DartLibLoader.loadLib
  package:dargon2/src/native/local_binder.dart 156:33            new LocalBinder
  package:dargon2/src/native/local_binder.dart 145:26            LocalBinder.instance
  package:dargon2/src/argon2.dart 86:37                          DArgon2.hashPasswordBytes
  package:dargon2/src/argon2.dart 52:12                          DArgon2.hashPasswordString

```

Tried: 

- unsign dart executable - no luck
- unsign signed macos dylib library - luck with a similar Error:

```
  Invalid argument(s): Failed to load dynamic library (dlopen(lib/src/blobs/libargon2-darwin.dylib, 1): no suitable image found. 
  dart:ffi                                                       new DynamicLibrary.open
  package:dargon2/src/native/loaders/dart_lib_loader.dart 20:27  DartLibLoader.loadLib
  package:dargon2/src/native/local_binder.dart 156:33            new LocalBinder
  package:dargon2/src/native/local_binder.dart 145:26            LocalBinder.instance
  package:dargon2/src/argon2.dart 86:37                          DArgon2.hashPasswordBytes
  package:dargon2/src/argon2.dart 52:12                          DArgon2.hashPasswordString

```

**Comment by tmthecoder** [comment]
Hey, I'm unable to replicate the error on mac, would you be able to let me know what you did if anything else than setup?

**Comment by tmthecoder** [comment]
@jayjah on that same note, i looked into the same error for a while, but only got the second load error, never the first one where it saw the package. I did manage to get that fixed, so if you try depending on dargon2: ^1.0.3, I think your error should be gone

**Comment by jayjah** [comment]
@tmthecoder Thank you for that fast response!
I will check it out later && I'll give you some feedback if it works!

**Comment by tmthecoder** [comment]
@jayjah No problem! Thanks for pointing the issue out!

**Comment by jayjah** [comment]
@tmthecoder Your fix 1.0.3 works just as intended! Thank you for that fast fix!
